hmt_escrow/eth_bridge.py

```diff
@@ -232,6 +238,100 @@ def deploy_factory(gas: int = GAS_LIMIT, **credentials) -> str:
+    in the hmt universe, defined by the kvstore key `hmt_pub_key`.  Works with the
+    `set_pub_key_at_address` function.
+
+    Requires that the `GAS_PAYER` environment variable be set to the
+    address that will be paying for the transaction on the ethereum network
+
+    Args:
+        wallet_addr (string): address to get the public key of
+
+    Returns:
+        bytes: the public key in bytes form
+
+    >>> import os
+    >>> from web3 import Web3
+    >>> get_pub_key_from_addr('badaddress')
+    Traceback (most recent call last):
+      File "/usr/lib/python3.6/doctest.py", line 1330, in __run
+        compileflags, 1), test.globs)
+      File "<doctest __main__.get_pub_key_from_addr[2]>", line 1, in <module>
+        get_pub_key_from_addr('blah')
+      File "hmt_escrow/eth_bridge.py", line 268, in get_pub_key_from_addr
+        raise ValueError('environment variable GAS_PAYER required')
+    ValueError: environment variable GAS_PAYER required
+    >>> os.environ['GAS_PAYER'] = "0x1413862C2B7054CDbfdc181B83962CB0FC11fD92"
+    >>> os.environ['GAS_PAYER_PRIV'] = "28e516f1e2f99e96a48a23cea1f94ee5f073403a1c68e818263f0eb898f1c8e5"
+    >>> set_pub_key_at_addr('blah')  #doctest: +ELLIPSIS
+    AttributeDict({'transactionHash': ...})
+    >>> get_pub_key_from_addr(os.environ['GAS_PAYER'])
+    b'blah'
+    """
+    # TODO: Should we try to get the checksum address here instead of assuming user will do that?
+    GAS_PAYER = os.getenv('GAS_PAYER')
+
+    if not GAS_PAYER:
+        raise ValueError('environment variable GAS_PAYER required')
+
+    w3 = get_w3()
+
+    kvstore = w3.eth.contract(address=KVSTORE_CONTRACT, abi=kvstore_abi)
+    addr_pub_key = kvstore.functions.get(GAS_PAYER, 'hmt_pub_key').call({
+        'from':
+        GAS_PAYER
+    })
+
+    return bytes(addr_pub_key, encoding='utf-8')
+
+
+def set_pub_key_at_addr(pub_key: str) -> Dict[str, Any]:
+    """
+    Given a public key, this function will use the eth-kvstore to reach out to the blockchain
+    and set the key `hmt_pub_key` on the callers kvstore collection of values, equivalent to the
+    argument passed in here.  This will be used by HMT to encrypt data for the receiver
+
+    See more about kvstore here: https://github.com/hCaptcha/eth-kvstore
+
+    Args:
+        pub_key (string): RSA Public key for this user
+
+    Returns:
+        AttributeDict: receipt of the set transaction on the blockchain
+
+
+    >>> from web3 import Web3
+    >>> import os
+    >>> os.environ['GAS_PAYER'] = "0x1413862C2B7054CDbfdc181B83962CB0FC11fD92"
+    >>> os.environ['GAS_PAYER_PRIV'] = "28e516f1e2f99e96a48a23cea1f94ee5f073403a1c68e818263f0eb898f1c8e5"
+    >>> pub_key_to_set = b"2dbc2c2c86052702e7c219339514b2e8bd4687ba1236c478ad41b43330b08488c12c8c1797aa181f3a4596a1bd8a0c18344ea44d6655f61fa73e56e743f79e0d"
+    >>> set_pub_key_at_addr(pub_key_to_set)  #doctest: +ELLIPSIS
+    AttributeDict({'transactionHash': ...})
+    """
```

Comment:
Nitpicking before perfect: one space needs to be added before closing docstring.

Reply:
Done but we're not aiming for perfection here folks ..